lib/mapObjects/IMarket.cpp

```diff
@@ -157,7 +157,7 @@ std::vector<int> IMarket::availableItemsIds(EMarketMode mode) const
 const IMarket * IMarket::castFrom(const CGObjectInstance *obj, bool verbose)
 {
 	auto * imarket = dynamic_cast<const IMarket *>(obj);
-	if(verbose && !imarket)
+	if(verbose && !imarket && obj)
```

Comment:
if obj is nullptr we simply ignore it?

Reply:
Oh, right. Should I split the error message in two?